src/daft-scan/src/lib.rs

```diff
@@ -31,6 +32,9 @@ use storage_config::StorageConfig;
 pub enum Error {
     #[cfg(feature = "python")]
     PyIO { source: PyErr },
+
+    #[snafu(display("Error when merging ScanTasks: {}", msg))]
```

Comment:
I would avoid using a catch all string for the error. You should have an error for each type of error you are creating the string for `differing PartitionSpecs `, `differing FileFormat`, etc. Otherwise we don't get any of the benefits of being able to "catch" the error

Reply:
Take a look at https://github.com/Eventual-Inc/Daft/blob/main/src/daft-parquet/src/lib.rs#L102 for example